xts/src/main/java/org/jboss/as/xts/XTSSubsystemParser.java

```diff
@@ -184,22 +188,41 @@ private void parseXTSEnvironmentElement(XMLExtendedStreamReader reader, ModelNod
      * @throws XMLStreamException
      */
     private void parseDefaultContextPropagationElement(XMLExtendedStreamReader reader, ModelNode subsystem) throws XMLStreamException {
-        final int count = reader.getAttributeCount();
-        for (int i = 0; i < count; i++) {
-            ParseUtils.requireNoNamespaceAttribute(reader, i);
-            final String value = reader.getAttributeValue(i);
-            final Attribute attribute = Attribute.forName(reader.getAttributeLocalName(i));
+        processAttributes(reader, (index, attribute) -> {
+            final String value = reader.getAttributeValue(index);
             switch (attribute) {
                 case ENABLED:
                     if (value == null || (!value.toLowerCase().equals("true") && !value.toLowerCase().equals("false"))) {
-                        throw ParseUtils.invalidAttributeValue(reader, i);
+                        throw ParseUtils.invalidAttributeValue(reader, index);
                     }
                     DEFAULT_CONTEXT_PROPAGATION.parseAndSetParameter(value, subsystem, reader);
                     break;
                 default:
-                    throw ParseUtils.unexpectedAttribute(reader, i);
+                    throw ParseUtils.unexpectedAttribute(reader, index);
             }
-        }
+        });
+
+        // Handle elements
+        ParseUtils.requireNoContent(reader);
+    }
+
+    /**
+     * Handle the async-registration element.
+     */
+    private void parseAsyncRegistrationElement(XMLExtendedStreamReader reader, ModelNode subsystem) throws XMLStreamException {
```

Comment:
You need to throw an error about an unexpected error if the namespace version is < 2.1